indsl/fluid_dynamics/friction.py

```diff
@@ -16,11 +18,332 @@ def Haaland(Re: pd.Series, roughness: float) -> pd.Series:
     the accuracy of the data.
 
     Args:
-        Re: Reynolds Number
-        roughness: Surface roughness
+        Re: Reynolds Number [-].
+        roughness: Surface roughness [m].
 
     Returns:
-        pandas.Series: Friction factor
+        pandas.Series: Friction factor [-]
     """
     den = -1.8 * np.log10((6.9 / Re[Re != 0]) + (roughness / 3.7) ** 1.1)
     return 1 / den**2
+
+
+@check_types
+def Colebrook(Re: Union[pd.Series, float], roughness_scaled: Union[pd.Series, float]) -> pd.Series:
+    r"""Colebrook approximation.
+
+    A computationally efficient Colebrook approximation taken from the following article
+
+     "Fast and Accurate Approximations for the Colebrook Equation"
+     Dag Biberg
+     Journal of Fluids Engineering Copyright VC 2017 by ASME MARCH 2017, Vol. 139 / 031401-1
+
+    This is a faster and more accurat approximation to Colebrook than Haaland
+
+    Args:
+        Re: Reynolds Number [-].
+        roughness_scaled: Scaled surface roughness [-].
+            The wall surface roughness is normally scaled with the pipe inner diameter
+
+    Returns:
+        pandas.Series: Friction factor [-]
+    """
+    from math import log as m_log
+
+    a = 2 / m_log(10)
+    b = 2.51
+    c = 3.7
+    ab = a * b
+
+    logReab = np.log(Re / ab)
```

Comment:
why use both math.log and np.log?

Reply:
math.log is 3-4 times faster than np.log for a single value calculation. But unless the input series is very short the effect on the total run time is negligible, so I agree it might not be the most important optimisation